### PR TITLE
Notifications

### PR DESCRIPTION
**Pull request by DanGodfrey**
The goal is to support our current and in development notifications. This directive will evolve as required to support new and different notifications.  

Can be tested here: https://dan-workspace-wealthsimple.fwd.wf/

The directive has the following inputs:
  type:  determines the font and background colours (danger | warning | info | success | neutral) [required]
  headerString: String displayed in the notification's header. [optional]
  bodyString:    String displayed in the notification's body. [required]
  actionString:  String displayed in the notification's button if actionable. [optional]
  actionable:     Flag indicating if the notification contains an action [optional: defaults to false]
  dismissible:   Flag indicating if the notification can be dismissed  [optional: defaults to false]
  action:           Function to trigger when clicking the button if actionable [optional]
  dismiss:        Function to trigger after hiding the notification if dismissible [optional]

Future development:
- Aesthetic overhaul as part of slingshot
- Primary vs Secondary buttons
- Custom HTML
- Multiple Notifications

### CODE REVIEW

**Comment by DanGodfrey** [comment]
cc @6

**Comment by DanGodfrey** [comment]
notifications-directive

File: src/layout/notifications/ws-notification.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  'use strict';
+
+  angular
+  .module('ws.common')
+  .directive('wsNotification', ['$templateCache', function ($templateCache) {
+    return {
+      restrict: 'E',
+      template: $templateCache.get('ws-notification.html'),
+      scope: {
+        // Allow for two-way binding of this variable
```
Comment (johnsintal) [inline src/layout/notifications/ws-notification.js]: comment

File: src/layout/notifications/ws-notification.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        type:         '@',
```
Comment (johnsintal) [inline src/layout/notifications/ws-notification.js]: it'd be nice to put a ng-doc about the types and what explain what these other attributes.

File: test/ws-notification.test.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    });
+  });
+
+  describe('with a header', function () {
+
+    injectElement('<ws-notification header-string="testheader" dismissible="false" actionable="false" type="neutral" body-string="test123"/>');
+
+    it('should render a header', function () {
+      expect(element.html()).toContain('test123');
+      expect(element.html()).toContain('h2');
```
Comment (johnsintal) [inline test/ws-notification.test.js]: nice directive tests!

**Comment by johnsintal** [comment]
:shipit:

**Pull request merged**